src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/WriteMavenPomStep.java

```diff
@@ -41,26 +44,31 @@
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.DataBoundSetter;
 
+import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 import javax.inject.Inject;
 import java.io.FileNotFoundException;
 import java.io.OutputStream;
 import java.io.Serializable;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Set;
 
 /**
  * Writes a maven pom file to the current working directory.
- *
  * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
  */
 public class WriteMavenPomStep extends Step {
 
     private String file;
-    private final Model model;
+    private final transient Model model;
```

Comment:
I ask again, why not just fix the issue with the object not being serialized?

Reply:
Well it would not be _successfully_ serialized in any event. At best we would have a marker noting that there is an unserializable parameter. Since this parameter contributes nothing to, e.g., Blue Ocean serialization (really the `DescriptorImpl` should be overriding `argumentsToString` to return just `file`), it may as well be `transient`.